### ISSUE TITLE
Packages need version pinning in install_shapeworks.sh/.bat

### ISSUE THREAD

**Issue by akenmorris** [SCIInstitute/ShapeWorks#1580]
See #1579 

others that need pinning to future proof this script

statsmodel
trimesh
shapely
nbstripout
ipyvtklink
itkwidgets

etc.

Also, we seem to have some incompatibilities on the ITK packages.  We have a mix of 5.0.1 and 5.2.1.post1 by the time the script completes.

itk                               5.0.1
itk-core                          5.2.1.post1
itk-filtering                     5.2.1.post1
itk-io                            5.0.1
itk-meshtopolydata                0.7.1
itk-numerics                      5.2.1.post1
itk-registration                  5.0.1
itk-segmentation                  5.0.1
itkwidgets                        0.32.0

We also have some packages being requested with a version and then later without a version (e.g. itkwidgets).

This is fairly critical as mistakes will render the release broken as the 6.1 release is now broken due to a pip upgrade.

**Comment by akenmorris** [comment]
Some tidbits from the conda install output:

```
Installing collected packages: param, jupyterlab-widgets, itk-core, traittypes, pyct, itk-numerics, ipywidgets, zstandard, itk-meshtopolydata, itk-filtering, ipympl, ipydatawidgets, colorcet, itkwidgets
  Attempting uninstall: itk-core
    Found existing installation: itk-core 5.0.1
    Uninstalling itk-core-5.0.1:
      Successfully uninstalled itk-core-5.0.1
  Attempting uninstall: itk-numerics
    Found existing installation: itk-numerics 5.0.1
    Uninstalling itk-numerics-5.0.1:
      Successfully uninstalled itk-numerics-5.0.1
  Attempting uninstall: itk-filtering
    Found existing installation: itk-filtering 5.0.1
    Uninstalling itk-filtering-5.0.1:
      Successfully uninstalled itk-filtering-5.0.1
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
itk 5.0.1 requires itk-core==5.0.1, but you have itk-core 5.2.1.post1 which is incompatible.
itk 5.0.1 requires itk-filtering==5.0.1, but you have itk-filtering 5.2.1.post1 which is incompatible.
itk 5.0.1 requires itk-numerics==5.0.1, but you have itk-numerics 5.2.1.post1 which is incompatible.
itk-segmentation 5.0.1 requires itk-filtering==5.0.1, but you have itk-filtering 5.2.1.post1 which is incompatible.
itk-registration 5.0.1 requires itk-filtering==5.0.1, but you have itk-filtering 5.2.1.post1 which is incompatible.
itk-io 5.0.1 requires itk-core==5.0.1, but you have itk-core 5.2.1.post1 which is incompatible.
Successfully installed colorcet-2.0.6 ipydatawidgets-4.2.0 ipympl-0.8.2 ipywidgets-7.6.5 itk-core-5.2.1.post1 itk-filtering-5.2.1.post1 itk-meshtopolydata-0.7.1 itk-numerics-5.2.1.post1 itkwidgets-0.32.0 jupyterlab-wi\
dgets-1.0.2 param-1.12.0 pyct-0.4.8 traittypes-0.2.1 zstandard-0.16.0
```

**Comment by akenmorris** [comment]
Is there a reason we can't use the itk 5.2.1.post1 version for all ITK packages?

**Comment by akenmorris** [comment]
More stuff:

```
The following packages will be DOWNGRADED:

  xorg-libx11                              1.7.2-h7f98852_0 --> 1.6.12-h36c2ea0_0
  xorg-libxext                             1.3.4-h7f98852_1 --> 1.3.4-h516909a_0
  xorg-libxpm                             3.5.13-h7f98852_0 --> 3.5.13-h516909a_0
  xorg-libxrender                      0.9.10-h7f98852_1003 --> 0.9.10-h516909a_1002
  xorg-libxt                               1.2.1-h7f98852_2 --> 1.1.5-h516909a_1003
```

**Comment by sheryjoe** [comment]
should we close this one? @akenmorris

**Comment by akenmorris** [comment]
Resolved in #1579
